syntax="proto3";

import "proto/users.proto";
import "google/protobuf/timestamp.proto";

package distributed_agenda;
option go_package = "./server/proto";

message Event{
    int64 id = 1;
    google.protobuf.Timestamp start = 2;
    google.protobuf.Timestamp end = 3;
    string name = 4;
    string description = 5;
    int64 groupId = 6;
    bool draft = 7;
}

message GetEventRequest{
    int64 id = 1;
}

message CreateEventRequest{
    Event event = 1;
}

message DeleteEventRequest{
    int64 id = 1;
}

message GetEventResponse{
    Event event = 1;
}

message CreateEventResponse{
    OperationOutcome result = 1;
    repeated string unavailable = 2;
}

message DeleteEventResponse{
    OperationOutcome result = 1;
}

message ConfirmEventRequest{
    int64 eventId = 1;
}

message ConfirmEventResponse{
    OperationOutcome result = 1;
}

message RejectEventRequest{
    int64 eventId = 1;
}

message RejectEventResponse{
    OperationOutcome result = 1;
}

service EventsService{

    rpc GetEvent(GetEventRequest) returns (GetEventResponse){}

    rpc CreateEvent(CreateEventRequest) returns (CreateEventResponse){}

    rpc DeleteEvent(DeleteEventRequest) returns (DeleteEventResponse){}

    rpc ConfirmEvent(ConfirmEventRequest) returns (ConfirmEventResponse){}

    rpc RejectEvent(RejectEventRequest) returns (RejectEventResponse){}
}